bin/mindexer

```diff
@@ -146,26 +149,27 @@ def main(
                 # index can't be used, no benefit over collection scan
                 benefit = 0
             else:
-                # different costs per unit of work depending if the query is covered or not
-                if query.is_covered(candidate):
-                    cost = IXSCAN_COST
-                else:
-                    cost = FETCH_COST
-
-                # estimate for number of "work units"
+                # estimate for number of documents needs to be fetched
                 est = get_estimate(fetch_query)
 
+                # estimate number of index key examined 
+                index_key_scanned = query.index_number_key_qurey(candidate)
+                index_key_scanned_est = get_estimate(index_key_scanned)
+
                 # if the query has a limit, and all fields of the filter are in the
                 # index, then we can cap the upper bound of the estimate at limit.
                 # if not, then the expected number of units of work to find all matches
                 # is equal to est. The expected case is equal to the worst case.
                 # See https://math.stackexchange.com/questions/2595408/hypergeometric-distribution-expected-number-of-draws-until-k-successes-are-dra
                 if query.limit is not None and query.is_subset(candidate):
                     est = min(query.limit, est)
+                    index_key_scanned_est = min(query.limit, index_key_scanned_est)
 
-                # benefit of the index over a collection scan (assuming cost of
-                # collscan = 1.0 relative to other costs)
-                benefit = estimator.get_cardinality() * COLLSCAN_COST - est * cost
+                # calculating index benefit by substract index cost from collection scan cost
```

Comment:
Can you update the comment and explain how the index_key_scanned_est factors in? Feel free to use multiple comment lines, more explicit is better :-)